par2/decoder.go

```diff
@@ -529,30 +529,35 @@ func (d *Decoder) Verify(checkParity bool) (bool, error) {
 		return false, errors.New("no parity data")
 	}
 
-	for _, info := range d.fileIntegrityInfos {
-		if !info.ok(d.sliceByteCount) {
-			return false, nil
-		}
-	}
-
-	for _, shard := range d.parityShards {
-		if shard == nil {
-			return false, nil
-		}
-	}
-
-	if !checkParity {
-		return true, nil
-	}
-
 	coder, dataShards, err := d.newCoderAndShards()
 	if err != nil {
 		return false, err
 	}
 
-	computedParityShards := coder.GenerateParity(dataShards)
-	eq := reflect.DeepEqual(computedParityShards, d.parityShards)
-	return eq, nil
+	ok, err := coder.CanReconstructData(dataShards, d.parityShards)
+
+	// for _, info := range d.fileIntegrityInfos {
```

Comment:
With the suggestions above, you can get rid of all the commented out code, I thnk